lua/apisix/admin/routes.lua

```diff
@@ -0,0 +1,114 @@
+local core = require("apisix.core")
+local schema_desc = core.json.decode([[{
+    "type": "object",
+    "properties": {
+        "methods": {
+            "type": "array",
+            "items": {
+                "type": "string",
+                "enum": ["GET", "PUT", "POST", "DELETE"]
+            }
+        },
+        "plugin_config": {
+            "type": "object"
+        },
+        "upstream": {
+            "type": "object",
+            "properties": {
+                "nodes": {
+                    "type": "object"
+                },
+                "type": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "nodes",
+                "type"
+            ]
+        },
+        "uri": {
+            "type": "string"
+        }
+    },
+    "required": [
```

Comment:
all required?

Reply:
I mainly used https://www.liquid-technologies.com/online-json-to-schema-converter to generate the schema context, it is hard to write the json schema by hand.